src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -74,28 +82,24 @@ public T DeserializePayload<T>(Message message)
         {
             T retValue = default(T);
 
-            // TODO: Currently we use json serializer auto only for non-testmessage types
-            // CHECK: Can't we just use auto for everything
-            if (MessageType.TestMessage.Equals(message.MessageType))
-            {
-                retValue = message.Payload.ToObject<T>();
-            }
-            else
-            {
-                retValue = message.Payload.ToObject<T>(serializer);
-            }
+            var versionedMessage = message as VersionedMessage;
+            var serializer = versionedMessage?.Version == 2 ? payloadSerializer2 : payloadSerializer;
 
+            retValue = message.Payload.ToObject<T>(serializer);
             return retValue;
         }
 
         /// <summary>
         /// Deserialize raw JSON to an object using the default serializer.
         /// </summary>
         /// <param name="json">JSON string.</param>
+        /// <param name="version">Version of serializer to be used.</param>
         /// <typeparam name="T">Target type to deserialize.</typeparam>
         /// <returns>An instance of <see cref="T"/>.</returns>
-        public T Deserialize<T>(string json)
+        public T Deserialize<T>(string json, int version = 1)
```

Comment:
Should the caller be aware of `version` of payload? How would it know, since the version is embedded inside the json?